### DISCUSSION TITLE
Dump Memray bin on interval

### DISCUSSION THREAD

**Discussion by iloveitaly** [bloomberg/memray#585] [Q&A] [answered]
I'm trying to track down a memory leak that results in a SIGKILL. Per the documentation, SIGKILL results in no output being dumped, which makes sense.

Is there a way to schedule a memray output dump at a specific interval? This way, if the process dies because of SIGKILL, I can still inspect memory usage.

**Comment by pablogsal** [comment, marked as answer]
Unless you are using aggregated files, memray dumps as it tracks allocations so even if you receive `SIGKILL` there will be a functional dump. With aggregated files the file is created at the end so in that case it won't work with `SIGKILL`

**Reply by pablogsal** [reply]
Check out https://bloomberg.github.io/memray/run.html#aggregated-capture-files

**Reply by iloveitaly** [reply]
Awesome, thank you!
